### PR TITLE
Remove search bar focus/blur behavior (alternative to #4876)

### PR DESCRIPTION
**Pull request by zbynek**
See #4818 and [JENKINS-56109](https://issues.jenkins-ci.org/browse/JENKINS-56109).

### Proposed changelog entries

* Make sure value does not disappear from search bar on blur

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@timja 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by timja — COMMENTED** [review]
Remove the class in pageHeader as well?

https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/layout/pageHeader.jelly#L53

**Review by timja — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 1]
```diff
@@ -728,20 +728,6 @@ function expandButton(e) {
     layoutUpdateCallback.call();
 }
 
-function inputHasDefaultTextOnFocus() {
```
Comment (fqueiruga) [inline war/src/main/webapp/scripts/hudson-behavior.js:731]: Did you make sure this function is not used elsewhere across the https://github.com/jenkinsci organization?

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 2]
```diff
@@ -728,20 +728,6 @@ function expandButton(e) {
...
-    if (this.value == defaultValue) {
-        this.value = "";
-        Element.removeClassName(this, "defaulted");
-    }
-}
-
-function inputHasDefaultTextOnBlur() {
```
Comment (fqueiruga) [inline war/src/main/webapp/scripts/hudson-behavior.js:738]: Ditto

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 3]
```diff
@@ -891,14 +877,6 @@ function rowvgStartEachRow(recursive,f) {
         makeButton(e, expandButton);
     });
 
-    // scripting for having default value in the input field
-    Behaviour.specify("INPUT.has-default-text", "input-has-default-text", ++p, function(e) {
```
Comment (fqueiruga) [inline war/src/main/webapp/scripts/hudson-behavior.js:895]: Same as above

File: war/src/main/webapp/scripts/hudson-behavior.js (hunk 1 above)
Comment (timja) [reply war/src/main/webapp/scripts/hudson-behavior.js:731]: not used https://github.com/search?q=org%3Ajenkinsci+inputHasDefaultTextOnFocus&type=Code

File: war/src/main/webapp/scripts/hudson-behavior.js (hunk 2 above)
Comment (timja) [reply war/src/main/webapp/scripts/hudson-behavior.js:738]: not used https://github.com/search?q=org%3Ajenkinsci+inputHasDefaultTextOnBlur&type=Code

File: war/src/main/webapp/scripts/hudson-behavior.js (hunk 3 above)
Comment (timja) [reply war/src/main/webapp/scripts/hudson-behavior.js:895]: not used https://github.com/search?q=org%3Ajenkinsci+has-default-text&type=Code

**Review by fqueiruga — APPROVED** [review]
I'm all for removing weird JS behaviour!

**Comment by timja** [comment]
This PR is ready for merge and may be merged after 24 hours if there's no negative feedback

**Pull request merged**